src/module/index.ts

```diff
@@ -2,6 +2,32 @@ import { resolve, join } from 'path'
 import merge from 'lodash/merge'
 import defaults from './defaults'
 
+function extendRoutes ({ pagesDepth }) {
+  function createRouteObject (component, depth:number, currentDepth:number = 0) {
+    if (currentDepth > depth) {
+      return null
+    }
+    const routeObject = {
+      name: 'page' + currentDepth,
+      path: ':page' + currentDepth + '?',
+      component,
+      children: null
+    }
+    if (currentDepth === 0) {
+      routeObject.path = '/' + routeObject.path
+    }
+    const child = createRouteObject(component, depth, currentDepth + 1)
+    if (child) {
+      routeObject.children = [child]
+    }
+    return routeObject
+  }
+  this.extendRoutes((routes) => {
+    const newRoutes = createRouteObject('~cwa/core/templates/page.vue', pagesDepth)
+    routes.push(newRoutes)
+  })
+}
+
 export default function (moduleOptions) {
   // Merge all option sources
   const options = merge({}, defaults, moduleOptions, this.options.auth)
```

Comment:
Why do we need the configuration from the auth module? Also, by doing this we may not get all the auth module options. For any module the options could also be defined in another way `this.options.modules[someIndex][1]` - can be seen here: https://nuxtjs.org/api/configuration-modules/